src/check_update.c

```diff
@@ -47,7 +47,7 @@ enum swupd_code check_update()
 	int current_version, server_version;
 	enum swupd_code ret;
 
-	ret = read_versions(&current_version, &server_version, globals.path_prefix);
+	ret = read_versions(&current_version, &server_version, globals.path_prefix, true);
```

Comment:
boolean parameters to modify how a function work is usually hard to understand. 
We can just use the get_current_version() and get_absolute_latest_version() here.